src/main/java/org/tron/core/db2/core/AbstractSnapshot.java

```diff
@@ -1,14 +1,13 @@
 package org.tron.core.db2.core;
 
-import java.lang.ref.Reference;
 import java.lang.ref.WeakReference;
 import lombok.Getter;
 import lombok.Setter;
 import org.tron.core.db2.common.DB;
 
 public abstract class AbstractSnapshot<K, V> implements Snapshot {
   @Getter
-  protected DB<K, V> db;
+  protected volatile DB<K, V> db;
   @Getter
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Remove the "volatile" keyword from this field. ([squid:S3077](https://sonarcloud.io/organizations/tron-zhaohong/rules#rule_key=squid%3AS3077))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=java-tron&pullRequest=1864&issues=AWfKoyOD_E3vci_sHyJ7&open=AWfKoyOD_E3vci_sHyJ7)

[ISSUE_KEY]: AWfKoyOD_E3vci_sHyJ7